app-shell/src/app/shell-parser/template-parser/parse5-template-parser.ts

```diff
@@ -3,6 +3,48 @@ import {TemplateParser} from './template-parser';
 
 var Parser = require('../../../vendor/parse5/lib/parser');
 var Serializer = require('../../../vendor/parse5/lib/serializer');
+var Tokenizer = require('../../../vendor/parse5/lib/tokenizer');
+
+// Monkey patching the lexer in order to establish
+// case sensitive parsing of the input templates.
+// This way we'll be able to use case sensitive attribute
+// and element selectors for stripping the content that
+// is not required for the App Shell.
+//
+// Since we're patching module's internals we cannot
+// use parse5 as dependency of the App Shell since we
+// won't have access to the tokenizer in order to patch
+// it runtime. Because of that we distribute the entire
+// Runtime Parser as a single bundle which includes parse5.
+Tokenizer.prototype.getNextToken = function () {
```

Comment:
I'm having trouble reading through this function (probably because it's end of day on a Friday :)). Can you add a few unit tests for this?